lightning/src/ln/channelmanager.rs

```diff
@@ -344,19 +346,37 @@ struct ReceiveError {
 /// using [`ChannelManager::fail_htlc_backwards_with_reason`].
 ///
 /// For more info on failure codes, see <https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages>.
+#[repr(u16)]
 #[derive(Clone, Copy)]
 pub enum FailureCode {
 	/// We had a temporary error processing the payment. Useful if no other error codes fit
 	/// and you want to indicate that the payer may want to retry.
-	TemporaryNodeFailure             = 0x2000 | 2,
+	TemporaryNodeFailure,
 	/// We have a required feature which was not in this onion. For example, you may require
 	/// some additional metadata that was not provided with this payment.
-	RequiredNodeFeatureMissing       = 0x4000 | 0x2000 | 3,
+	RequiredNodeFeatureMissing,
 	/// You may wish to use this when a `payment_preimage` is unknown, or the CLTV expiry of
 	/// the HTLC is too close to the current block height for safe handling.
 	/// Using this failure code in [`ChannelManager::fail_htlc_backwards_with_reason`] is
 	/// equivalent to calling [`ChannelManager::fail_htlc_backwards`].
-	IncorrectOrUnknownPaymentDetails = 0x4000 | 15,
+	IncorrectOrUnknownPaymentDetails,
+	/// We failed to process the payload after the onion was decrypted. You may wish to
+	/// use this when receiving custom HTLC TLVs with even type numbers that you don't recognize.
+	///
+	/// The tuple data should include the type number and byte offset in the decrypted byte stream
+	/// where the failure occurred.
+	InvalidOnionPayload(Option<(u64, u16)>),
```

Comment:
So the user won't really have the ability to know the byte offset, but the type number could be helpful for the sender to debug. In most places where we return this error code we don't specify any data at all - wasn't sure if it makes sense to include this `Option<(u64, u16)>` in there or not?